### PR TITLE
feat: Edm4hep refactoring

### LINKED ISSUE #1: refactor IRT `EVALUATION` for the `edm4hep` update
Since the data model has changed, we need to refactor some of the `evaluation` code. Things to do include:
- update our usage of `eic::Index`, since we now use PODIO relations instead
- should we use `edm4hep::MCParticle` to replace `dd4pod::Geant4Particle`?
- how to match `ReconstructedParticle` to the MC generated particle? we no longer have `mcID`, do we need to add a relation in `eicd`?

### PR DESCRIPTION
**Pull request by c-dilks**
close #1

### CODE REVIEW

**Comment by c-dilks** [comment]
@chchatte92 I moved your `reader_2.C` from `drich-dev` to here in `scripts/`. I had to make the following change to make it work:
```diff
192c192,193
<         particle->PIDReconstruction(pid, photons);
---
>         for(auto photon : photons) particle->FindRadiatorHistory(gas)->AddOpticalPhoton(photon);
>         particle->PIDReconstruction(pid);
```
Do you intend for this `reader_2.C` to replace the original `scripts/reader.C`? I think we should minimize the number of duplicated/similar scripts.

**Comment by c-dilks** [comment]
The code in the `evaluation/` subdirectory is still broken, but I don't think that's a priority to fix at the moment, since the `scripts/` macros are more up-to-date, and AFAIK they do about the same thing. We could open an issue for this, but better to resolve #6 first.

**Comment by chchatte92** [comment]
Yes. Please go ahead.

File: scripts/evaluation.fig12.C [hunk 1]
```diff
@@ -68,14 +68,14 @@ void evaluation(const char *ifname, const char *ofname = 0)
...
     for(const auto &rctrack: *rctracks) 
       mc2rc[rctrack.mcID] = &rctrack;
 #endif
     
     // Then the Cherenkov-to-reconstructed mapping; FIXME: may want to use Cherenkov-to-simulated 
     // mapping to start with, for the debugging purposes;
-    std::map<eic::Index, const eic::CherenkovParticleIDData*> rc2cherenkov;
+    std::map<eic::Index, const edm4eic::CherenkovParticleIDData*> rc2cherenkov;
     for(const auto &pid: *cherenkov) 
       rc2cherenkov[pid.recID] = &pid;
```
Comment (c-dilks) [inline scripts/evaluation.fig12.C:80]: this script is also likely broken, and will need the same refactoring as `evaluation.C` and `reader.C` needed

File: scripts/rejection.C [hunk 2]
```diff
@@ -69,14 +69,14 @@ void rejection(const char *ifname)//, const char *ofname = 0)
 
 #ifdef _USE_RECONSTRUCTED_TRACKS_
     // First populate the reconstructed-to-simulated particle mapping table;
-    std::map<eic::Index, const eic::ReconstructedParticleData*> mc2rc;
+    std::map<eic::Index, const edm4eic::ReconstructedParticleData*> mc2rc;
     for(const auto &rctrack: *rctracks) 
       mc2rc[rctrack.mcID] = &rctrack;
```
Comment (c-dilks) [inline scripts/rejection.C:74]: also probably broken

**Pull request merged**